cpp/include/raft/neighbors/ivf_flat_types.hpp

```diff
@@ -278,26 +345,29 @@ struct index : ann::index {
   uint32_t veclen_;
   raft::distance::DistanceType metric_;
   bool adaptive_centers_;
-  device_mdarray<T, extent_2d<IdxT>, row_major> data_;
-  device_mdarray<IdxT, extent_1d<IdxT>, row_major> indices_;
-  device_mdarray<uint32_t, extent_1d<uint32_t>, row_major> list_sizes_;
-  device_mdarray<IdxT, extent_1d<uint32_t>, row_major> list_offsets_;
-  device_mdarray<float, extent_2d<uint32_t>, row_major> centers_;
-  std::optional<device_mdarray<float, extent_1d<uint32_t>, row_major>> center_norms_;
+  bool conservative_memory_allocation_;
+  host_vector<std::shared_ptr<list_data<T, IdxT>>, uint32_t> lists_;
```

Comment:
```suggestion
  std::vector<std::shared_ptr<list_data<T, IdxT>>, uint32_t> lists_;
```

see discussion https://github.com/rapidsai/raft/pull/1249#discussion_r1107267407